src/ServiceControl/CustomChecks/SqlDbCustomCheckDataStore.cs

```diff
@@ -0,0 +1,212 @@
+namespace ServiceControl.CustomChecks
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Data.SqlClient;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Threading.Tasks;
+    using Contracts.CustomChecks;
+    using Contracts.Operations;
+    using Dapper;
+    using Infrastructure;
+    using Infrastructure.DomainEvents;
+    using Infrastructure.Extensions;
+    using Lucene.Net.Analysis.Tokenattributes;
+    using Raven.Database.Storage.Voron.Impl;
+
+    class SqlDbCustomCheckDataStore : ICustomChecksStorage
+    {
+        public SqlDbCustomCheckDataStore(string connectionString, IDomainEvents domainEvents, EndpointDetailsMapper mapper)
+        {
+            this.connectionString = connectionString;
+            this.domainEvents = domainEvents;
+            this.mapper = mapper;
+        }
+
+        public async Task UpdateCustomCheckStatus(CustomCheckDetail detail)
+        {
+            var publish = false;
```

Comment:
I wonder if this flag (and the details of the custom check) should be returned to the caller. That way all of the publishing logic can happen outside of the storage.

Reply:
I have extracted the custom check status notification from the DataStore. It should address this comment, please review the commit [1ff24d8](https://github.com/Particular/ServiceControl/pull/2915/commits/1ff24d8439a8e192583f5bc97755faa809c5a44b).